File: Common/RMA.Common/Entities/Note.cs
using System;

namespace RMA.Common.Entities
{
    public class Note
    {
        public int Id { get; set; }
        public string ItemType { get; set; }
        public int ItemId { get; set; }
        public string Text { get; set; }
        public string Reason { get; set; }
        public bool IsDeleted { get; set; }
        public string CreatedBy { get; set; }
        public DateTime CreatedDate { get; set; }
        public string ModifiedBy { get; set; }
        public DateTime ModifiedDate { get; set; }
    }
}

File: Web/RMA-Web-Shell/projects/shared-components-lib/src/lib/notes/note.ts
export class Note  {
    id: number;
    itemType: string;
    itemId: number;
    text: string;
    reason: string;

    isDeleted: boolean;
    createdBy: string;
    modifiedBy: string;
    createdDate: Date;
    modifiedDate: Date;
    isActive: boolean;

    public static copy(note: Note): Note
    {
        let newNote = new Note();
        newNote.itemType = note.itemType;
        newNote.itemId = note.itemId;
        newNote.text = note.text;
        newNote.reason = note.reason;
        newNote.isDeleted = note.isDeleted;
        return newNote;
    }
}
